framework/src/media/audio/audio_manager.c

```diff
@@ -648,8 +651,8 @@ static audio_manager_result_t set_audio_volume(audio_io_direction_t direct, uint
 	}
 
 	ret = AUDIO_MANAGER_SUCCESS;
-	config->volume = volume;
-	medvdbg("Volume = %d\n", volume);
+	config->volume = caps_desc.caps.ac_controls.hw[0];
```

Comment:
scaled_volume

Reply:
Now I do the volume scaling in each set_input/output_audio_gain/volume() after this function is finished.

Reply:
No, I mean there's little chance that modify caps_desc.caps.ac_controls.hw[0] in device driver, using scaled_volume is correct.

Reply:
Oh, I understand. Modified.